media/server/gstplayer/source/GstGenericPlayer.cpp

```diff
@@ -630,6 +648,41 @@ bool GstGenericPlayer::setCodecData(GstCaps *caps, const std::shared_ptr<CodecDa
     return false;
 }
 
+bool GstGenericPlayer::pushSampleIfRequired(GstElement *source, GstBuffer *buffer)
+{
+    auto initialPosition = m_context.initialPositions.find(source);
+    if (m_context.initialPositions.end() == initialPosition)
+    {
+        return false;
+    }
+    RIALTO_SERVER_LOG_DEBUG("Pushing new sample...");
+    GstSegment *segment{m_gstWrapper->gstSegmentNew()};
+    m_gstWrapper->gstSegmentInit(segment, GST_FORMAT_TIME);
+    if (!m_gstWrapper->gstSegmentDoSeek(segment, m_context.playbackRate, GST_FORMAT_TIME, GST_SEEK_FLAG_NONE,
+                                        GST_SEEK_TYPE_SET, initialPosition->second, GST_SEEK_TYPE_SET,
+                                        GST_CLOCK_TIME_NONE, nullptr))
+    {
+        RIALTO_SERVER_LOG_WARN("Segment seek failed.");
+        m_gstWrapper->gstSegmentFree(segment);
+        m_context.initialPositions.erase(initialPosition);
+        return false;
+    }
+
+    RIALTO_SERVER_LOG_INFO("New segment: [%" GST_TIME_FORMAT ", %" GST_TIME_FORMAT "], rate: %f \n",
```

Comment:
Maybe change to milestone. It's worth to have important seek logs as milestone. They shouldn't flood us and might be helpful.

Reply:
severity changed